Add jonmorton-haven

jeongseok-meta opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/haven/meta.yaml```) and found it was in an excellent condition.

jeongseok-meta commented:
@conda-forge/help-python, ready for review!

synapticarbors commented on recipes/haven/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+
+requirements:
+  build:
+    - flit-core
```
Typically, I don't think we need `flit-core` under `build`

synapticarbors commented on recipes/haven/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  commands:
+    - pip check
+  requires:
+    - numpy
```
Do we need `numpy` here?

synapticarbors commented on recipes/haven/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  host:
+    - pip
+    - python {{ python_min }}
+    - setuptools
```
Is `setuptools` needed if flit is being used as the build backend?

synapticarbors commented on recipes/haven/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+{% set name = "haven" %}
```
This overlaps with other packages on `pypi`. I'm not sure how best to deal with this.

jeongseok-meta replied:
Yeah, I've also noticed that. Perhaps augment it with the organization name, such as `jonmorton-haven`?

jeongseok-meta replied:
Renamed to `jonmorton-haven`

synapticarbors commented on recipes/haven/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  version: {{ version }}
+
+source:
+  url: https://github.com/jonmorton/haven/archive/2e501f2db2557bf62cd69c86e2fac1b2280e5770.tar.gz
```
This url is likely to cause issues for our regro bot. It would be good if there was an official tagged release up on github.

jeongseok-meta replied:
Unfortunately, the upstream repository does not have a tag. An issue has been created for this: https://github.com/jonmorton/haven/issues/11

jeongseok-meta commented:
@synapticarbors I've addressed your comments, and this PR is now ready for another review. It's renamed to `jonmorton-haven` to avoid the potential name conflict, and I have added a comment explaining why a tag is not used (it does not exist yet). Thank you!

jeongseok-meta commented:
@synapticarbors sorry for direct ping. Could you please take another look for this PR when you have a chance? Thanks!

jeongseok-meta commented:
@synapticarbors  have you had a chance to take another look at this PR? I've renamed the package to `jonmorton-haven` (respecting the author's name) to avoid the name conflict.

jeongseok-meta commented:
@conda-forge-admin Could anyone help review this PR? Thanks!

jeongseok-meta commented:
@conda-forge/core Could anyone help review this PR? Thanks!
